src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -177,9 +178,33 @@ public Response update(@Auth AuthUser authUser, @Context UriInfo info, @PathPara
             // Ensure that we have a real user with this ID, fail if we do not.
             userService.findUserById(userId);
             URI uri = info.getRequestUriBuilder().path("{id}").build(userId);
-            User user = userService.updateUserFieldsById(userUpdateFields, userId);
+            User updatedUser = userService.updateUserFieldsById(userUpdateFields, userId);
+            Gson gson = new Gson();
+            JsonObject jsonUser = userService.findUserWithPropertiesByIdAsJsonObject(authUser, updatedUser.getUserId());
+            return Response.ok(uri).entity(gson.toJson(jsonUser)).build();
+        } catch (Exception e) {
+            return createExceptionResponse(e);
+        }
+    }
+
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @PermitAll
+    public Response updateSelf(@Auth AuthUser authUser, @Context UriInfo info, String json) {
+        try {
+            User user = userService.findUserByEmail(authUser.getEmail());
+            UserUpdateFields userUpdateFields = gson.fromJson(json, UserUpdateFields.class);
+
+            if (Objects.nonNull(userUpdateFields.getUserRoleIds()) && !user.hasUserRole(UserRoles.ADMIN)) {
+                throw new BadRequestException("Cannot change user roles");
+            }
+
+            user = userService.updateUserFieldsById(userUpdateFields, user.getUserId());
             Gson gson = new Gson();
             JsonObject jsonUser = userService.findUserWithPropertiesByIdAsJsonObject(authUser, user.getUserId());
+
+            URI uri = info.getRequestUriBuilder().path("").build();
```

Comment:
I know this is copied from the other `PUT` method, but both are unnecessary. This is supposed to build a URL to return with the response, but it's not working as expected in both cases. The actual URL should look like host + `/api/user`, but in both cases, it does not. I suggest we remove it from both methods.